src/Handler/Order/UpdatePaymentMethodHandler.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\ShopApiPlugin\Handler\Order;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\OrderPaymentStates;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Core\Repository\PaymentMethodRepositoryInterface;
+use Sylius\ShopApiPlugin\Command\Order\UpdatePaymentMethod;
+use Webmozart\Assert\Assert;
+
+final class UpdatePaymentMethodHandler
+{
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var PaymentMethodRepositoryInterface */
+    private $paymentMethodRepository;
+
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        PaymentMethodRepositoryInterface $paymentMethodRepository
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+
+    public function __invoke(UpdatePaymentMethod $choosePaymentMethod): void
+    {
+        /** @var OrderInterface $order */
+        $order = $this->orderRepository->findOneBy(['tokenValue' => $choosePaymentMethod->orderToken()]);
+
+        Assert::notNull($order, 'Order has not been found.');
+        Assert::notSame(OrderInterface::STATE_CART, $order->getState(), 'Only orders can be updated.');
```

Comment:
Not 100% true, I suppose. Order's in fulfilled state should be blocked as well. 

```suggestion
        Assert::same(OrderInterface::STATE_NEW, $order->getState(), 'Only new orders can be updated.');
```

Reply:
The validator checks if the payment you are trying to edit was not payed yet. This would be a safer option as an assertion?